vivarium/core/process.py

```diff
@@ -143,29 +151,157 @@ def __init__(self, parameters: Optional[dict] = None) -> None:
         elif not hasattr(self, 'name'):
             self.name = self.__class__.__name__
 
-        self.parameters = copy.deepcopy(self.defaults)
-        self.parameters = deep_merge(self.parameters, parameters)
-        self.schema_override = self.parameters.pop('_schema', {})
-        self.parallel = self.parameters.pop('_parallel', False)
-        self.condition_path = None
+        self._parameters = copy.deepcopy(self.defaults)
+        self._parameters = deep_merge(self._parameters, parameters)
+        self._schema_override: Schema = self._parameters.pop('_schema', {})
+        self._parallel = self._parameters.pop('_parallel', False)
+        self._condition_path: Optional[HierarchyPath] = None
+        self._command_result: Any = None
 
         # set up the conditional state if a condition key is provided
-        if '_condition' in self.parameters:
-            self.condition_path = self.parameters.pop('_condition')
-        if self.condition_path:
-            self.merge_overrides(assoc_path({}, self.condition_path, {
+        if '_condition' in self._parameters:
+            self._condition_path = self._parameters.pop('_condition')
+        if self._condition_path:
+            self.merge_overrides(assoc_path({}, self._condition_path, {
                 '_default': True,
                 '_emit': True,
                 '_updater': 'set'}))
 
         self._set_timestep()
 
-        self.schema: Optional[dict] = None
+        self._schema: Optional[Schema] = None
+
+    @property
+    def parameters(self) -> dict:
+        return self._parameters
+
+    @property
+    def schema_override(self) -> Schema:
+        return self._schema_override
+
+    @property
+    def parallel(self) -> bool:
+        return self._parallel
+
+    @property
+    def condition_path(self) -> Optional[HierarchyPath]:
+        return self._condition_path
+
+    @property
+    def schema(self) -> Optional[Schema]:
+        return self._schema
```

Comment:
just to make sure I understand -- these `@property` will allow parallel processes to be inspected by calling these methods rather than directly inspecting the variables (which are now private)

Reply:
can you call `properties` without the parentheses? `self.parameters()` would be a pretty big breaking change.

Reply:
Reading these properties works exactly like before. If you have a process `p`, then `p.schema` will evaluate to the result of calling the `schema()` method of `p`. (I'm not sure `p.schema()` will even work). That should mean there's no breaking change

Reply:
Interesting, I haven't used these before. I try to avoid special features like this, but this is a good use case really.